src/main/java/at/ac/tuwien/kr/alpha/config/AlphaContext.java

```diff
@@ -0,0 +1,27 @@
+package at.ac.tuwien.kr.alpha.config;
+
+/**
+ * Wrapper type for AlphaConfig and InputConfig.
+ */
+public class AlphaContext {
```

Comment:
I dislike that we end up with `AlphaContext` and `AlphaConfig`, which might be confusing. Also, why is it a context and not just another config? You might well have `AlphaConfig` containing `InputConfig` and `SystemConfig`. Or have `AlphaConfig` contain `InputConfig`?

Reply:
The idea behind calling this AlphaContext was that it wraps the 2 configs and is the top-level "config-like" structure, similar to an ApplicationContext in Spring.
AlphaConfig shouldn't directly contain InputConfig - The idea is that if you were to use Alpha as an API, you'd have one Alpha instance (initialized with an AlphaConfig) that represents the system. Whenever you want to solve a program (from whatever source), you'd create an InputConfig and pass it to your Alpha instance. In short, the separation between the two configs makes sure Alpha doesn't have an internal state that depends on any specific program.
However, I like your naming suggestion, so I'll rename the current AlphaConfig to SystemConfig and AlphaContext to AlphaConfig

Reply:
Makes sense. Thank you for elaborating. At least we don't mix `Context` and `Config` now.